src/markdown/parsers/cardParser.ts

```diff
@@ -135,12 +135,16 @@ export class CardParser extends BaseParser {
    * @private
    */
   async linesToHtml(lines: string[]) {
+    const fixLatex = (match: string) => (
+      match.replace(/\\[{}%#&$_\\]/g, str => str === "\\\\" ? "\\\\\\\\" : "\\" + str)
+    );
     const string = lines.join("\n")
       // $$\{1,2\} \%100$$ => $$\\{1,2\\} \\%100$$
-      .replace(/(?<!\\)\$\$\n?(.+?)\n?(?<!\\)\$\$/g, str => str.replace(/(\\[{}%#&$_])/g, "\\$1"))
+      .replace(/(?<!\\)\$\$\n?.+?\n?(?<!\\)\$\$/g, fixLatex)
       // $\{1,2\} \%100$ => $\\{1,2\\} \\%100$
-      .replace(/(?<![\\$])\$(?!\$)(.+?)(?<!\\)\$/g, str => str.replace(/(\\[{}%#&$_])/g, "\\$1"));
+      .replace(/(?<![\\$])\$(?!\$).+?(?<!\\)\$/g, fixLatex);
```

Comment:
Do we need tests for this?, It would be great if you were able to add some as I'm not sure what's being fixed here..

Reply:
Something like.. https://github.com/jasonwilliams/anki/blob/main/src/markdown/__tests__/Serializer.test.ts for an example

Reply:
when I write sth like `$\{1,2\}$` or `$\%100$` in md file, [marked](https://github.com/markedjs/marked) will regard `\` as escape character, and then remove it. please see [this issue](https://github.com/markedjs/marked/issues/546).

for example: `$a=\%100$`. When executing to [this code](https://github.com/jasonwilliams/anki/blob/c0114a2df8fb329069e6a445fde0feff85293e02/src/markdown/parsers/cardParser.ts#L140), we will get `<p>$a=%100$</p>` rather than `<p>$a=\%100$</p>` .`\` is missing. However, we can make the conversion correct by adding another `\` in front of it.

So the function of these codes is to add a `\` before these characters (\%->\\% \&->\\&) in latex context.

Reply:
I have tested several markdown parser, the result is same. Seems like that ignoring backslash before a char is a supposed behavior cause it is considered as escaping the char.